nfpm.go

```diff
@@ -30,12 +36,66 @@ func Get(format string) (Packager, error) {
 	return p, nil
 }
 
+// Parse decodes YAML data from an io.Reader into a configuration struct
+func Parse(in io.Reader) (config Config, err error) {
+	dec := yaml.NewDecoder(in)
+	dec.SetStrict(true)
+	err = dec.Decode(&config)
+	return
+}
+
+// ParseFile decodes YAML data from a file path into a configuration struct
+func ParseFile(path string) (config Config, err error) {
+	var file *os.File
+	file, err = os.Open(path)
+	if err != nil {
+		return
+	}
+	defer file.Close() // nolint: errcheck
+	return Parse(file)
+}
+
 // Packager represents any packager implementation
 type Packager interface {
 	Package(info Info, w io.Writer) error
 }
 
-// Info contains information about the package
+// Config contains the top level configuration for packages
+type Config struct {
```

Comment:
maybe we could do something like:

```go
type Overrideables struct {
  Depends []string `yaml:"depends"`
  // etc
} 

type Info struct {
  // Some fields
  Overrideables `yaml:",inline"`
  Overrides map[string]Overrideables `yaml:"overrides,omitempty"`
```

What do you think?

Reply:
That is a really nice way to do it. It all depends on whether or not we will use per-package fields (as is my plan right now). I would also like strict parsing of overrides (e.g. *illegal override* errors when applicable). Another approach:

```go
type Info struct {
  Depends []string `yaml:"depends" nfpm:"override"`
  // etc
}
```
I like both. Hopefully I'll arrive at something after tonight.